stories/widgets.stories.js

```diff
@@ -131,3 +169,31 @@ storiesOf( 'Global/Widgets', module )
 			) ) }
 		</CompositeWidgets>
 	) );
+
+storiesOf( 'Global/Widgets/Widget Area', module )
+	.addDecorator( ( storyFn ) => storyFn( createTestRegistry() ) )
+	.add( 'Regular sizes', ( registry ) => (
+		<WithTestRegistry registry={ registry }>
+			{ createWidgetAreas(
+				registry,
+				[ QUARTER, QUARTER, QUARTER, QUARTER ],
+				[ HALF, QUARTER, QUARTER ],
+				[ QUARTER, HALF, QUARTER ],
+				[ QUARTER, QUARTER, HALF ],
+				[ HALF, HALF ],
+				[ FULL ],
+			) }
+		</WithTestRegistry>
+	) )
+	.add( 'Irregular sizes', ( registry ) => (
+		<WithTestRegistry registry={ registry }>
+			{ createWidgetAreas(
+				registry,
+				[ QUARTER, QUARTER, HALF, [ QUARTER, FULL ] ],
+				[ [ HALF, FULL ], QUARTER, QUARTER ],
+				[ QUARTER, [ FULL, HALF ], QUARTER ],
```

Comment:
This test case is great to have, I wanna point this one out particularly because right now it is not showing up the way it should. Per the ACs and my other comment above, the following should happen here:
1. Render quarter widget (1).
2. Notice that the full widget (2) doesn't fit into the row (`counter` would be 15), hence try to see if the widget (2) has a smaller size that fits - which it does, so use the half width version of (2) instead.
3. Render the quarter widget (3), which will now actually even fit into the same row.

Reply:
It should be fixed now after amending the algorithm.